### PR TITLE
Added Authentication to Flask-Admin

### PR DESCRIPTION
**Pull request by rochacbruno**

### CODE REVIEW

File: app/models.py [hunk 1]
```diff
@@ -176,3 +178,63 @@ def fullname(self):
...
+    confirmed_at = db.Column(db.DateTime())
+    roles = db.relationship('Role', secondary=roles_users,
+                            backref=db.backref('users', lazy='dynamic'))
+
+    def __str__(self):
+        return self.email
+
+    @property
+    def is_root(self):
+        # this logic can be changed, ideas?
```
Comment (rochacbruno) [inline app/models.py]: @omaciel currently you have to use 'register' button to register yourself with one of the emails below to become root. But that logic can be changed. The better way would be using the `python manage.py createsuperuser` command. But it is marked as `TODO:`

File: config/development.py [hunk 2]
```diff
@@ -6,3 +6,22 @@
...
+SECURITY_LOGOUT_URL = "/logout/"
+SECURITY_REGISTER_URL = "/register/"
+
+SECURITY_POST_LOGIN_VIEW = "/admin/"
+SECURITY_POST_LOGOUT_VIEW = "/admin/"
+SECURITY_POST_REGISTER_VIEW = "/admin/"
+
+# Flask-Security features
+SECURITY_REGISTERABLE = True
+SECURITY_SEND_REGISTER_EMAIL = False
```
Comment (rochacbruno) [inline config/development.py:27]: The SECURITY_\* vars above may be moved to `production` and `testing` config files once gets stable. take a look at Flask-Security docs if something needs to be changed

File: app/__init__.py [hunk 3]
```diff
@@ -2,8 +2,10 @@
 
 from flask import Flask
 from flask_bootstrap import Bootstrap
+from flask_security import Security, SQLAlchemyUserDatastore
```
Comment (omaciel) [inline app/__init__.py:5]: Need to add dep to `requirements.txt`?

File: app/models.py (hunk 1 above)
Comment (omaciel) [inline app/models.py]: @rochacbruno so anyone trying to register with one of these 2 emails will automatically became a superuser, right? hmmm can't deploy it in a public place yet then :)

File: requirements.txt [hunk 4]
```diff
@@ -3,3 +3,4 @@ Flask-Admin
 Flask-Bootstrap
 Flask-SQLAlchemy
 Flask-WTF
+Flask-Security>=1.7.5
```
Comment (rochacbruno) [inline requirements.txt:6]: @omaciel Flask-Security is already here!

File: app/models.py (hunk 1 above)
Comment (rochacbruno) [inline app/models.py]: @omaciel I'll implement the `python manage.py createsuperuser` than root users can be created by command line.

Users are unique, so if registered once by you, no one can duplicate it.

File: app/__init__.py (hunk 3 above)
Comment (rochacbruno) [reply app/__init__.py:5]: @omaciel it is there on requirements

File: app/models.py (hunk 1 above)
Comment (rochacbruno) [reply app/models.py]: @omaciel updated! removed that `is_root` logic, now you have to run `python manage.py createsuperuser` in order to create root users.

**Review by omaciel — APPROVED** [review]
Thank you!

**Pull request merged**